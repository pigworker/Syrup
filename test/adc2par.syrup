!<Bit> -> <Bit>
!X = Z where
  Z = nand(X, X)

<Bit> & <Bit> -> <Bit>
X & Y = Z where
  A = nand(X, Y)
  Z = !A

<Bit> | <Bit> -> <Bit>
X | Y = Z where
  A, B = !X, !Y
  Z = nand(A, B)

nor(<Bit>, <Bit>) -> <Bit>
nor(X,Y) = !(X | Y)

xor(<Bit>, <Bit>) -> <Bit>
xor(X, Y) = Z where
  A, B = X | Y, X & Y
  C = !B
  Z = A & C

hadd(<Bit>, <Bit>) -> <Bit>, <Bit>
hadd(X1, Y1) = C2, S1 where
  C2 = X1 & Y1
  S1 = xor(X1, Y1)

fadd(<Bit>, <Bit>, <Bit>) -> <Bit>, <Bit>
fadd(X1, Y1, C1) = C2, Z1 where
  D2, D1 = hadd(X1, Y1)
  E2, Z1 = hadd(D1, C1)
  C2 = xor(D2, E2)

adc2([<Bit>,<Bit>],[<Bit>,<Bit>],<Bit>) ->
  <Bit>,[<Bit>,<Bit>]
adc2([X2,X1],[Y2,Y1],C1) = C4,[Z2,Z1] where
  C2,Z1 = fadd(X1,Y1,C1)
  C4,Z2 = fadd(X2,Y2,C2)

one() -> <Bit>
one() = !zero()

mux(<Bit>, <Bit>, <Bit>) -> <Bit>
mux(C, X, Y) = C & X | !C & Y

adc2par([<Bit>,<Bit>],[<Bit>,<Bit>],<Bit>) ->
  <Bit>,[<Bit>,<Bit>]
adc2par([X2,X1],[Y2,Y1],C1) = C4,[Z2,Z1] where
  C2,Z1   = fadd(X1,Y1,C1)
  C40,Z20 = fadd(X2,Y2,zero())
  C41,Z21 = fadd(X2,Y2,one())
  C4=mux(C2,C41,C40)
  Z2=mux(C2,Z21,Z20)

display adc2par