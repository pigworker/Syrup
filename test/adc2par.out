Circuit not is defined.

Circuit and is defined.

Circuit or is defined.

Circuit nor is defined.

Circuit xor is defined.

Circuit hadd is defined.

Circuit fadd is defined.

Circuit adc2 is defined.

Circuit one is defined.

Circuit mux is defined.

Circuit adc2par is defined.

<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
 "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.43.0 (0)
 -->
<!-- Title: whitebox Pages: 1 -->
<svg width="342pt" height="672pt"
 viewBox="0.00 0.00 342.00 672.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 668)">
<title>whitebox</title>
<polygon fill="white" stroke="transparent" points="-4,4 -4,-668 338,-668 338,4 -4,4"/>
<g id="clust1" class="cluster">
<title>cluster_circuit__</title>
</g>
<!-- GATE_adc2par_13__INPUTS -->
<g id="node1" class="node">
<title>GATE_adc2par_13__INPUTS</title>
<text text-anchor="start" x="93" y="-632" font-family="Times,serif" font-size="20.00">[X2, X1]</text>
<text text-anchor="start" x="189" y="-632" font-family="Times,serif" font-size="20.00">[Y2, Y1]</text>
<text text-anchor="start" x="283" y="-632" font-family="Times,serif" font-size="20.00">C1</text>
</g>
<!-- GATE_fadd_13__7__9 -->
<g id="node9" class="node">
<title>GATE_fadd_13__7__9</title>
<text text-anchor="start" x="251" y="-476.4" font-family="Times,serif" font-size="7.00">X1</text>
<text text-anchor="start" x="272" y="-476.4" font-family="Times,serif" font-size="7.00">Y1</text>
<text text-anchor="start" x="292" y="-476.4" font-family="Times,serif" font-size="7.00">C1</text>
<text text-anchor="start" x="253" y="-450" font-family="Times,serif" font-size="20.00">fadd</text>
<text text-anchor="start" x="251" y="-430.4" font-family="Times,serif" font-size="7.00">C2</text>
<text text-anchor="start" x="272" y="-430.4" font-family="Times,serif" font-size="7.00">Z1</text>
<polygon fill="none" stroke="black" points="242,-421 242,-489 310,-489 310,-421 242,-421"/>
</g>
<!-- GATE_adc2par_13__INPUTS&#45;&gt;GATE_fadd_13__7__9 -->
<g id="edge12" class="edge">
<title>GATE_adc2par_13__INPUTS:C1&#45;&gt;GATE_fadd_13__7__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M297,-623C297,-563.46 297,-547.27 297,-490.27"/>
<polygon fill="black" stroke="black" stroke-width="2" points="298.75,-490 297,-485 295.25,-490 298.75,-490"/>
<text text-anchor="middle" x="299.5" y="-547.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- FANOUT_13__8 -->
<g id="node10" class="node">
<title>FANOUT_13__8</title>
<polygon fill="red" stroke="transparent" points="234,-555 224,-555 224,-548 234,-548 234,-555"/>
</g>
<!-- GATE_adc2par_13__INPUTS&#45;&gt;FANOUT_13__8 -->
<g id="edge14" class="edge">
<title>GATE_adc2par_13__INPUTS:__VIRTUAL__7&#45;&gt;FANOUT_13__8:n</title>
<path fill="none" stroke="black" stroke-width="4" d="M229,-623C229,-593.44 229,-586.06 229,-556.5"/>
<text text-anchor="middle" x="231.5" y="-580.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- FANOUT_13__9 -->
<g id="node11" class="node">
<title>FANOUT_13__9</title>
<polygon fill="red" stroke="transparent" points="139,-555 129,-555 129,-548 139,-548 139,-555"/>
</g>
<!-- GATE_adc2par_13__INPUTS&#45;&gt;FANOUT_13__9 -->
<g id="edge13" class="edge">
<title>GATE_adc2par_13__INPUTS:__VIRTUAL__6&#45;&gt;FANOUT_13__9:n</title>
<path fill="none" stroke="black" stroke-width="4" d="M134,-623C134,-593.44 134,-586.06 134,-556.5"/>
<text text-anchor="middle" x="136.5" y="-580.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- FANIN_13__0 -->
<g id="node2" class="node">
<title>FANIN_13__0</title>
<polygon fill="red" stroke="transparent" points="210,-116 200,-116 200,-109 210,-109 210,-116"/>
</g>
<!-- GATE_adc2par_13__OUTPUTS -->
<g id="node15" class="node">
<title>GATE_adc2par_13__OUTPUTS</title>
<text text-anchor="start" x="112" y="-22" font-family="Times,serif" font-size="20.00">C4</text>
<text text-anchor="start" x="154" y="-22" font-family="Times,serif" font-size="20.00">[Z2, Z1]</text>
</g>
<!-- FANIN_13__0&#45;&gt;GATE_adc2par_13__OUTPUTS -->
<g id="edge7" class="edge">
<title>FANIN_13__0:s&#45;&gt;GATE_adc2par_13__OUTPUTS:__VIRTUAL__8</title>
<path fill="none" stroke="black" stroke-width="4" d="M205,-107.5C205,-79.36 196.14,-71.76 195.1,-46.02"/>
<polygon fill="black" stroke="black" stroke-width="4" points="196.85,-45.96 195,-41 193.35,-46.03 196.85,-45.96"/>
<text text-anchor="middle" x="205.5" y="-75.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_mux_13__1__12 -->
<g id="node3" class="node">
<title>GATE_mux_13__1__12</title>
<text text-anchor="start" x="200.5" y="-222.4" font-family="Times,serif" font-size="7.00">C</text>
<text text-anchor="start" x="221" y="-222.4" font-family="Times,serif" font-size="7.00">X</text>
<text text-anchor="start" x="239.5" y="-222.4" font-family="Times,serif" font-size="7.00">Y</text>
<text text-anchor="start" x="200" y="-196" font-family="Times,serif" font-size="20.00">mux</text>
<polygon fill="none" stroke="black" points="188.5,-175 188.5,-235 253.5,-235 253.5,-175 188.5,-175"/>
</g>
<!-- GATE_mux_13__1__12&#45;&gt;FANIN_13__0 -->
<g id="edge21" class="edge">
<title>GATE_mux_13__1__12:s&#45;&gt;FANIN_13__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M203,-179C203,-151.67 203,-144.83 203,-117.5"/>
<text text-anchor="middle" x="205.5" y="-141.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_mux_13__2__12 -->
<g id="node4" class="node">
<title>GATE_mux_13__2__12</title>
<text text-anchor="start" x="105.5" y="-222.4" font-family="Times,serif" font-size="7.00">C</text>
<text text-anchor="start" x="126" y="-222.4" font-family="Times,serif" font-size="7.00">X</text>
<text text-anchor="start" x="144.5" y="-222.4" font-family="Times,serif" font-size="7.00">Y</text>
<text text-anchor="start" x="105" y="-196" font-family="Times,serif" font-size="20.00">mux</text>
<polygon fill="none" stroke="black" points="93.5,-175 93.5,-235 158.5,-235 158.5,-175 93.5,-175"/>
</g>
<!-- GATE_mux_13__2__12&#45;&gt;GATE_adc2par_13__OUTPUTS -->
<g id="edge22" class="edge">
<title>GATE_mux_13__2__12:s&#45;&gt;GATE_adc2par_13__OUTPUTS:C4</title>
<path fill="none" stroke="black" stroke-width="2" d="M108,-179C108,-118.96 124.96,-103.73 125.95,-46.32"/>
<polygon fill="black" stroke="black" stroke-width="2" points="127.71,-46.01 126,-41 124.21,-45.98 127.71,-46.01"/>
<text text-anchor="middle" x="121.5" y="-108.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_one_13__3__11 -->
<g id="node5" class="node">
<title>GATE_one_13__3__11</title>
<text text-anchor="start" x="168" y="-454" font-family="Times,serif" font-size="20.00">one</text>
<polygon fill="none" stroke="black" points="160.5,-433 160.5,-477 211.5,-477 211.5,-433 160.5,-433"/>
</g>
<!-- GATE_fadd_13__4__9 -->
<g id="node6" class="node">
<title>GATE_fadd_13__4__9</title>
<text text-anchor="start" x="141" y="-349.4" font-family="Times,serif" font-size="7.00">X1</text>
<text text-anchor="start" x="162" y="-349.4" font-family="Times,serif" font-size="7.00">Y1</text>
<text text-anchor="start" x="182" y="-349.4" font-family="Times,serif" font-size="7.00">C1</text>
<text text-anchor="start" x="143" y="-323" font-family="Times,serif" font-size="20.00">fadd</text>
<text text-anchor="start" x="141" y="-303.4" font-family="Times,serif" font-size="7.00">C2</text>
<text text-anchor="start" x="162" y="-303.4" font-family="Times,serif" font-size="7.00">Z1</text>
<polygon fill="none" stroke="black" points="132,-294 132,-362 200,-362 200,-294 132,-294"/>
</g>
<!-- GATE_one_13__3__11&#45;&gt;GATE_fadd_13__4__9 -->
<g id="edge23" class="edge">
<title>GATE_one_13__3__11:s&#45;&gt;GATE_fadd_13__4__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M187,-437C187,-403.67 187,-394.07 187,-363.15"/>
<polygon fill="black" stroke="black" stroke-width="2" points="188.75,-363 187,-358 185.25,-363 188.75,-363"/>
<text text-anchor="middle" x="189.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_fadd_13__4__9&#45;&gt;GATE_mux_13__1__12 -->
<g id="edge16" class="edge">
<title>GATE_fadd_13__4__9:s&#45;&gt;GATE_mux_13__1__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M167,-298C167,-279.51 163.11,-271.94 174,-257 187.61,-238.34 218.62,-253.28 223.38,-236.1"/>
<polygon fill="black" stroke="black" stroke-width="2" points="225.13,-236.17 224,-231 221.66,-235.75 225.13,-236.17"/>
<text text-anchor="middle" x="176.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_fadd_13__4__9&#45;&gt;GATE_mux_13__2__12 -->
<g id="edge15" class="edge">
<title>GATE_fadd_13__4__9:s&#45;&gt;GATE_mux_13__2__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M146,-298C146,-269.08 130.93,-262.46 129.17,-236.15"/>
<polygon fill="black" stroke="black" stroke-width="2" points="130.91,-235.94 129,-231 127.41,-236.05 130.91,-235.94"/>
<text text-anchor="middle" x="142.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_zero_13__5__2 -->
<g id="node7" class="node">
<title>GATE_zero_13__5__2</title>
<text text-anchor="start" x="79" y="-454" font-family="Times,serif" font-size="20.00">zero</text>
<polygon fill="none" stroke="black" points="71.5,-433 71.5,-477 130.5,-477 130.5,-433 71.5,-433"/>
</g>
<!-- GATE_fadd_13__6__9 -->
<g id="node8" class="node">
<title>GATE_fadd_13__6__9</title>
<text text-anchor="start" x="43" y="-349.4" font-family="Times,serif" font-size="7.00">X1</text>
<text text-anchor="start" x="64" y="-349.4" font-family="Times,serif" font-size="7.00">Y1</text>
<text text-anchor="start" x="84" y="-349.4" font-family="Times,serif" font-size="7.00">C1</text>
<text text-anchor="start" x="45" y="-323" font-family="Times,serif" font-size="20.00">fadd</text>
<text text-anchor="start" x="43" y="-303.4" font-family="Times,serif" font-size="7.00">C2</text>
<text text-anchor="start" x="64" y="-303.4" font-family="Times,serif" font-size="7.00">Z1</text>
<polygon fill="none" stroke="black" points="34,-294 34,-362 102,-362 102,-294 34,-294"/>
</g>
<!-- GATE_zero_13__5__2&#45;&gt;GATE_fadd_13__6__9 -->
<g id="edge24" class="edge">
<title>GATE_zero_13__5__2:s&#45;&gt;GATE_fadd_13__6__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M102,-437C102,-413.29 115.36,-406.54 108,-384 104.18,-372.31 92.89,-372.08 89.8,-363.16"/>
<polygon fill="black" stroke="black" stroke-width="2" points="91.49,-362.68 89,-358 88.03,-363.21 91.49,-362.68"/>
<text text-anchor="middle" x="112.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_fadd_13__6__9&#45;&gt;GATE_mux_13__1__12 -->
<g id="edge18" class="edge">
<title>GATE_fadd_13__6__9:s&#45;&gt;GATE_mux_13__1__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M69,-298C69,-266.79 97.53,-269.78 126,-257 172.5,-236.12 237.26,-280.51 241.75,-236.2"/>
<polygon fill="black" stroke="black" stroke-width="2" points="243.51,-236.08 242,-231 240.01,-235.91 243.51,-236.08"/>
<text text-anchor="middle" x="128.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_fadd_13__6__9&#45;&gt;GATE_mux_13__2__12 -->
<g id="edge17" class="edge">
<title>GATE_fadd_13__6__9:s&#45;&gt;GATE_mux_13__2__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M48,-298C48,-277.53 53.47,-270.34 69,-257 95.31,-234.4 141.95,-264.2 146.62,-236.04"/>
<polygon fill="black" stroke="black" stroke-width="2" points="148.37,-236.12 147,-231 144.88,-235.85 148.37,-236.12"/>
<text text-anchor="middle" x="71.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- GATE_fadd_13__7__9&#45;&gt;FANIN_13__0 -->
<g id="edge20" class="edge">
<title>GATE_fadd_13__7__9:s&#45;&gt;FANIN_13__0:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M277,-425C277,-312.06 317.64,-272.94 269,-171 253.33,-138.15 207,-153.9 207,-117.5"/>
<text text-anchor="middle" x="294.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__12 -->
<g id="node14" class="node">
<title>COPY_13__12</title>
<polygon fill="skyblue" stroke="transparent" points="232,-331.5 222,-331.5 222,-324.5 232,-324.5 232,-331.5"/>
</g>
<!-- GATE_fadd_13__7__9&#45;&gt;COPY_13__12 -->
<g id="edge19" class="edge">
<title>GATE_fadd_13__7__9:s&#45;&gt;COPY_13__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M256,-425C256,-382.13 227,-375.87 227,-333"/>
<text text-anchor="middle" x="253.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- FANOUT_13__8&#45;&gt;GATE_fadd_13__7__9 -->
<g id="edge8" class="edge">
<title>FANOUT_13__8:s&#45;&gt;GATE_fadd_13__7__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M231,-546.5C231,-514.23 272.11,-518.52 276.6,-490.25"/>
<polygon fill="black" stroke="black" stroke-width="2" points="278.37,-490.12 277,-485 274.88,-489.85 278.37,-490.12"/>
<text text-anchor="middle" x="262.5" y="-514.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__10 -->
<g id="node12" class="node">
<title>COPY_13__10</title>
<polygon fill="skyblue" stroke="transparent" points="50,-458.5 40,-458.5 40,-451.5 50,-451.5 50,-458.5"/>
</g>
<!-- FANOUT_13__8&#45;&gt;COPY_13__10 -->
<g id="edge9" class="edge">
<title>FANOUT_13__8:s&#45;&gt;COPY_13__10:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M227,-546.5C227,-522.4 207.95,-520.96 186,-511 133.29,-487.07 102,-529.42 57,-493 44.87,-483.18 45,-475.61 45,-460"/>
<text text-anchor="middle" x="220.5" y="-514.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- FANOUT_13__9&#45;&gt;GATE_fadd_13__7__9 -->
<g id="edge10" class="edge">
<title>FANOUT_13__9:s&#45;&gt;GATE_fadd_13__7__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M136,-546.5C136,-520.68 159.31,-523.33 182,-511 210.96,-495.27 251.29,-517.2 255.62,-490.19"/>
<polygon fill="black" stroke="black" stroke-width="2" points="257.38,-490.11 256,-485 253.89,-489.86 257.38,-490.11"/>
<text text-anchor="middle" x="184.5" y="-514.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__11 -->
<g id="node13" class="node">
<title>COPY_13__11</title>
<polygon fill="skyblue" stroke="transparent" points="26,-458.5 16,-458.5 16,-451.5 26,-451.5 26,-458.5"/>
</g>
<!-- FANOUT_13__9&#45;&gt;COPY_13__11 -->
<g id="edge11" class="edge">
<title>FANOUT_13__9:s&#45;&gt;COPY_13__11:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M132,-546.5C132,-496.49 65.68,-530.86 33,-493 22.8,-481.19 21,-475.61 21,-460"/>
<text text-anchor="middle" x="126.5" y="-514.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__10&#45;&gt;GATE_fadd_13__4__9 -->
<g id="edge2" class="edge">
<title>COPY_13__10:s&#45;&gt;GATE_fadd_13__4__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M47,-450C47,-410.36 72.14,-402.87 107,-384 130.86,-371.08 162.46,-384.95 166.56,-363.2"/>
<polygon fill="black" stroke="black" stroke-width="2" points="168.32,-363.13 167,-358 164.83,-362.83 168.32,-363.13"/>
<text text-anchor="middle" x="109.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__10&#45;&gt;GATE_fadd_13__6__9 -->
<g id="edge1" class="edge">
<title>COPY_13__10:s&#45;&gt;GATE_fadd_13__6__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M43,-450C43,-426.11 53.65,-422.03 60,-399 64.45,-382.84 68.17,-377.68 68.88,-363.27"/>
<polygon fill="black" stroke="black" stroke-width="2" points="70.63,-363.04 69,-358 67.13,-362.96 70.63,-363.04"/>
<text text-anchor="middle" x="66.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__11&#45;&gt;GATE_fadd_13__4__9 -->
<g id="edge4" class="edge">
<title>COPY_13__11:s&#45;&gt;GATE_fadd_13__4__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M23,-450C23,-416.59 31.37,-402.78 59,-384 90.81,-362.38 141.02,-395.02 145.65,-363.18"/>
<polygon fill="black" stroke="black" stroke-width="2" points="147.41,-363.11 146,-358 143.92,-362.87 147.41,-363.11"/>
<text text-anchor="middle" x="61.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__11&#45;&gt;GATE_fadd_13__6__9 -->
<g id="edge3" class="edge">
<title>COPY_13__11:s&#45;&gt;GATE_fadd_13__6__9:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M19,-450C19,-419.82 24.63,-412.35 35,-384 38.75,-373.76 45.43,-371.49 47.42,-363.27"/>
<polygon fill="black" stroke="black" stroke-width="2" points="49.19,-363.16 48,-358 45.71,-362.78 49.19,-363.16"/>
<text text-anchor="middle" x="37.5" y="-387.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__12&#45;&gt;GATE_mux_13__1__12 -->
<g id="edge6" class="edge">
<title>COPY_13__12:s&#45;&gt;GATE_mux_13__1__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M229,-323C229,-282.34 205.19,-274.16 203.14,-236.29"/>
<polygon fill="black" stroke="black" stroke-width="2" points="204.88,-235.95 203,-231 201.38,-236.04 204.88,-235.95"/>
<text text-anchor="middle" x="214.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
<!-- COPY_13__12&#45;&gt;GATE_mux_13__2__12 -->
<g id="edge5" class="edge">
<title>COPY_13__12:s&#45;&gt;GATE_mux_13__2__12:n</title>
<path fill="none" stroke="black" stroke-width="2" d="M225,-323C225,-307.67 223.33,-302.86 215,-290 208.92,-280.61 186.61,-262.73 177,-257 150.5,-241.2 112.74,-261.15 108.41,-236.2"/>
<polygon fill="black" stroke="black" stroke-width="2" points="110.14,-235.85 108,-231 106.65,-236.12 110.14,-235.85"/>
<text text-anchor="middle" x="198.5" y="-260.8" font-family="Times,serif" font-size="14.00"> </text>
</g>
</g>
</svg>

